### ISSUE TITLE
Feasibility of using runtime generated animation.

### ISSUE THREAD

**Issue by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [EsotericSoftware/spine-runtimes#2265]
Scenario,
I wanted to create game* with dynamic animation.In my mind I can think of 2 way.
- manually doing multiple animation,eg when get hit on head,hit on body,hit on knee,hit on toe,etc...
- Generate/Calculate in real time,feed the values/generated animation to spine-runtime

Reason why generate animation.
Say if I have 8 character,to animate and I plan 5 animation,hit head,body,knee,toe&idle. 
Simple math tells me I need to draw 40 animation.Which is quite tedious.
Not only that the extra problem is that its VERY not extensible.
If I need to add 2 more character say player parents,we need to do 10 more animations.

However,if we use generated dynamic animation.
8 character,we can just do idle animation.
8 animation +1 code to generate/calculate animation.
Also,This has the added benefits of linear extension,need a new character sure!
Just create a simple a spine and a idle animation,DONE!

Using horrible* naive math to measure benefits:10/50 animation,80% cost reduction.

Assume I can put some effort to homeroll/copy paste code for IK,Interpolation,some realistic/cartoonish physics math
- I think spine-runtime has interpolation built in I however didnt check if its exposed like an api or we can make it auto does the IK constraint for the whole character,

so....Here is my core question.
___________
TLDR

1,is it possible to make spine-runtime perform animation/changes using delta changes(json/some code* collection object)?
{"head":[-1,-0.333],"neck":[-0.8,-0.22]}
or java.util.Map? c# System.ValueTuple?

2,does the api for spine runtimes functions and can we use it?
say if there is a interpolation functions in spine-runtime.can we do something like.
>character1.Interpolate(assume_delta_animationjson_possible,spine.EInterpolcation.LINEARINTERPOLATE,speed=2f);

3,if 1 is possible can we somehow queue/chain animation?
eg we 1st call.
>character1.Interpolate(character_gethit_physics_calculated_falling,spine.EInterpolcation.LINEARINTERPOLATE,speed=1f);
then
character1.Interpolate(character_try_standupanim,spine.EInterpolcation.LINEARINTERPOLATE,speed=1f);
then
character1.Interpolate(character_holdgun_aimfront,spine.EInterpolcation.SINEWAVEINTERPOLATE,speed=1f);

I decided to ask here instead of spending more time aimlessly searching and thinking myself.
I believe experienced dev worked with spine probably know its possible or not more than an outsider.

Any infos is appreciated,Thank you.
